test/test_endtoend.py

```diff
@@ -161,6 +173,9 @@ def _run_server(self):
             args.append('--log-level')
             args.append(logging.getLevelName(log_level).lower())
 
+        for extra_arg in extra_args:
```

Comment:
I thought adding two lists would create a new list, which would be somewhat meaningless when an empty extra_args is given (Creating a new PyListObject, pointing names to new PyObject...), and I thought it would be efficient to actually have the append done inside PyListObejct. It seems like += is equivalent to extend(), so I have fixed this.